grpc-proto/proto/query.proto

```diff
@@ -195,118 +265,250 @@ message TypeSpec {
+  // operation.
   int32 block_for = 3;
+  // The number of replicas that experienced a failure while executing the request.
   int32 num_failures = 4;
+  // Whether the actual data was amongst the received replica responses.
+  // During reads, Cassandra doesn't request data from every replica to minimize internal network
+  // traffic. Instead, some replicas are only asked for a checksum of the data. A read timeout may
+  // occur even if enough replicas have responded to fulfill the consistency level, if only checksum
+  // responses have been received. This field allows to detect that case.
   bool data_present = 5;
 };
 
+// An error during the execution of a CQL function.
 message FunctionFailure {
+  // The keyspace containing the function.
   string keyspace = 1;
+  // The name of the function.
   string function = 2;
+  // The CQL types of the arguments of the function.
   repeated string arg_types = 3;
 }
 
+// A non-timeout error during a write query.
+// This happens when some of the replicas that were contacted by the coordinator replied with an
+// error.
 message WriteFailure {
+  // The consistency level of the operation that failed.
   Consistency consistency = 1;
+  // The number of replica that had acknowledged/responded to the operation before it failed.
   int32 received = 2;
+  // The minimum number of replica acknowledgements/responses that were required to fulfill the
+  // operation.
   int32 block_for = 3;
+  // The number of replicas that experienced a failure while executing the request.
   int32 num_failures = 4;
+  // The type of the write for which an error was raised.
   string write_type = 5;
 }
 
+// Thrown when a query attempts to create a keyspace or table that already exists.
 message AlreadyExists {
   string keyspace = 1;
   string table = 2;
 }
 
+// An exception occured due to a contended Compare And Set write/update.
+// The CAS operation was only partially completed and the operation may or may not get completed by
+// the contending CAS write or SERIAL/LOCAL_SERIAL read.
 message CasWriteUnknown {
+  // The consistency level of the operation that failed.
   Consistency consistency = 1;
+  // The number of replica that had acknowledged/responded to the operation before it failed.
   int32 received = 2;
+  // The minimum number of replica acknowledgements/responses that were required to fulfill the
+  // operation.
   int32 block_for = 3;
 }
 
+// A single row in a result set.
 message Row {
+  // The values for the columns (in the same order as ResultSet.columns).
   repeated Value values = 1;
 }
 
+// The data returned in response to a CQL query.
 message ResultSet {
+  // Metadata about the columns that this result set contains.
+  // This may be omitted if explicitly requested by the client (see QueryParameters.skip_metadata
+  // and BatchParameters.skip_metadata).
   repeated ColumnSpec columns = 1;
+
+  // The rows of CQL data.
   repeated Row rows = 2;
+
+  // The number of rows.
```

Comment:
I made a mistake. I think `page_size` should not be in `Response`.  This is redundant; the number of row can be determined from `rows`.

Reply:
https://github.com/stargate/stargate/issues/1161